src/components/Map/SearchPane/Drawer/DrawerContent/EntityDetails/EntityComponents/EntityEmissionsData/EntityEmissionsData.tsx

```diff
@@ -5,112 +5,184 @@ import EntityPostsList from './BottomViews/EntityPosts/EntityPostsList';
+      );
+      const electricityData = sortedData.filter(
+        (item: any) =>
+          item.measure === "electricity" &&
+          new Date(item.period_end).getFullYear() === year
+      );
+
+      let yearEmissions = sumEmissions(gasData);
+      yearEmissions += sumEmissions(electricityData);
+
+      if (yearEmissions > 0) {
+        totalEmissions = yearEmissions;
+        yearOfData = year;
+        return true;
+      }
+    });
+    if (totalEmissions === 0 && sortedData.length) {
+      const mostRecentYear = new Date(sortedData[0].period_end).getFullYear();
+      const mostRecentYearData = sortedData.filter(
+        (item: any) =>
+          new Date(item.period_end).getFullYear() === mostRecentYear
+      );
+
+      yearOfData = mostRecentYear;
+      totalEmissions += sumEmissions(mostRecentYearData);
+    }
+
+    return {
+      year: yearOfData,
+      emissions: Math.round(totalEmissions).toString(),
     };
+  };
 
-    useEffect(() => {
-        if(graphData.length === 0 && actionData.length === 0 && postData.length === 0) {
-            props.setIsEmpty(true);
-        } else {
-            props.setIsEmpty(false);
-        }
-    }, [graphData, actionData, postData])
-    
-    useEffect(() => {
-        if(props.emissionsData && props.entitiesByBusinessType.length === 0) {
-            getEntityData();
-        }
-    }, []);
-    
-    return (
-        <div className='accordion-list'>
-            {graphData.length === 0 && actionData.length === 0 && postData.length === 0 ?
-                <div className="empty-entity-data">
-                    <p>This organisation hasn't submitted any data yet.</p>
-                </div>
-            :
-            <>
-                <DataAccordion 
-                    title="CO2e in 2021"
-                    titleData="24t"
-                    bottomView={
-                        <EntityCO2 
-                            name={props.emissionsData[0].name}
-                            graphData={graphData}
-                            labels={graphData.map((emission:any) => emission.measure)}
-                            data={graphData.map((emission:any) => emission.value)}
-                        />
-                    }
-                />
-                <DataAccordion 
-                    title="actions"
-                    titleData={actionData.length}
-                    bottomView={
-                        <EntityActionsList 
-                            actions={actionData}
-                        />
-                    }
-                />
-                <DataAccordion 
-                    title="posts"
-                    titleData={postData.length} 
-                    bottomView={
-                        <EntityPostsList 
-                            posts={postData}
-                        />
-                    }
-                />
-            </>
-            }
+  const { year, emissions } = checkFullYear();
```

Comment:
It's technically possible for year to come back undefined from checkFullYear() for a few circumstances where there are zero emissions. It's not likely but check how the code copes with that - can't quite do the logic in my head!